app/controllers/concerns/error_handling.rb

```diff
@@ -3,13 +3,17 @@
 module ErrorHandling
   extend ActiveSupport::Concern
 
+  # rubocop:disable Metrics/BlockLength
   included do
     class UnknownOrder < RuntimeError; end
     class BadRequest < RuntimeError; end
 
     rescue_from StandardError do |exception|
-      render json: { errors: "Server encountered an unexpected error: #{exception.inspect}" },
-             status: :internal_server_error
+      render :json => exception_hash(
+        exception,
+        status: :bad_request,
+        detail: "Server encountered an unexpected error: #{exception.inspect}"
+      ), status: :bad_request
```

Comment:
Shouldn't this be kept as `:internal_server_error`?

Reply:
My bad. Probably copy-paste error. I'll fix it.